src/badge/badge.service.ts

```diff
@@ -35,4 +41,11 @@ export class BadgeService {
       .where('badge.id = :badgeId', { badgeId: id })
       .getMany();
   }
+
+  async grantBadgeJob(badgeId: string, playerId: string): Promise<any> {
```

Comment:
Call it `grantBadge`. This method should give a badge to the player, and then add the job to the queue.